docs/llm-hosting/bench/grammartest.py

```
@@ -0,0 +1,102 @@
+        "messages": [{"role": "user", "content": make_prompt(nwords, int(time.time_ns() % 10**9))}],
+        "max_tokens": GEN, "temperature": 0.7, "stream": True,
+        "stream_options": {"include_usage": True},
+    }
+    if with_tools:
+        payload["tools"] = TOOLS
+        payload["tool_choice"] = "auto"
+    req = json.dumps(payload).encode()
+    t0 = time.perf_counter()
+    ttft, n, usage = None, 0, {}
+    r = urllib.request.Request(URL, data=req, headers={"Content-Type": "application/json"})
+    with urllib.request.urlopen(r, timeout=1800) as resp:
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)